src/main.rs

```diff
@@ -1,4 +1,143 @@
-fn main() {
-    println!("Hello, world!");
+use axum::{
+    async_trait,
+    extract::{FromRequest, Path, RequestParts},
+    handler::Handler,
+    http::StatusCode,
+    response::{IntoResponse, Json, Response},
+    routing::get,
+    Router, Server,
+};
+use serde_json::Value;
+use std::collections::HashMap;
+
+#[derive(Debug)]
+enum Version {
+    V1,
+}
+
+#[derive(Debug)]
+struct JsonProxyError(reqwest::Error);
+
+#[tokio::main]
+async fn main() {
+    let meta_routes = Router::new()
+        .route("/", get(root_handler))
+        .route("/ping", get(heartbeat_handler));
+
+    let transit_routes = Router::new()
+        .route("/bus", get(transit_bus_handler))
+        .route("/modes", get(transit_modes_handler));
+
+    let api_routes = Router::new()
+        .route("/contacts", get(contacts_handler))
+        .route("/dictionary", get(dictionary_handler))
+        .route("/faqs", get(faqs_handler))
+        .route("/food/named/menu/pause", get(pause_menu_handler))
+        .route("/printing/color-printers", get(color_printers_handler))
+        .route("/spaces/hours", get(hours_handler))
+        .route("/tools/help", get(help_handler))
+        .nest("/transit", transit_routes)
+        .route("/webcams", get(webcams_handler));
+
+    let app = Router::new()
+        .nest("/", meta_routes)
+        .nest("/api/:version", api_routes)
+        .fallback(fallback.into_service());
+
+    Server::bind(&"0.0.0.0:3000".parse().unwrap())
+        .serve(app.into_make_service())
+        .await
+        .expect("server failed to exit successfully");
+}
+
+#[allow(clippy::unused_async)]
+async fn root_handler() -> Result<&'static str, StatusCode> {
+    Ok("Hello world!")
+}
+
+#[allow(clippy::unused_async)]
+async fn heartbeat_handler() -> Result<&'static str, StatusCode> {
+    Ok("pong")
+}
+
+#[allow(clippy::unused_async)]
+async fn fallback() -> impl IntoResponse {
+    (StatusCode::NOT_FOUND, "Not Found".to_string())
+}
+
+async fn gh_pages_handler(filename: &str) -> Result<Json<Value>, JsonProxyError> {
+    let url = format!("https://stodevx.github.io/AAO-React-Native/{}", filename).to_string();
+    let resp = request_handler(&url).await?;
+    Ok(resp)
 }
 
+macro_rules! gh_pages_handler {
+    ($name:ident,$filename:literal) => {
+        async fn $name(_version: Version) -> Result<Json<Value>, JsonProxyError> {
+            let data = gh_pages_handler($filename).await.unwrap();
+            Ok(data)
+        }
+    };
+}
```

Comment:
In order to make a generic version of `gh_pages_handler` that deserializes to a particular type (e.g. some mythical type like `ContactInfoResponse`), you'll probably want another version of this macro that also takes a `ty` argument.

You can do this by just adding another `(..) => {..};` definition to this `macro_rules` block:

```rust
($name:ident,$filename:literal,$response_type:ty) => {
    async fn $name(_version: Version) -> Result<Json<$response_type>, JsonProxyError> {
        let data = gh_pages_handler($filename).await?;
        Ok(data)
    }
};
```

(The only appreciable difference here is the `$response_type:ty` argument, and the change to the `Json<_>` argument to explicitly pass it.)